kv/main.go

```diff
@@ -158,25 +152,25 @@ type cockroach struct {
 	writeStmt *sql.Stmt
 }
 
-func (c *cockroach) read(blockID int64) error {
-	var writerID string
-	var rawBytes []byte
-	if err := c.readStmt.QueryRow(blockID).Scan(&writerID, &rawBytes); err != nil && err != sql.ErrNoRows {
+func (c *cockroach) read(k int64) error {
+	var v []byte
+	if err := c.readStmt.QueryRow(k).Scan(&v); err != nil && err != sql.ErrNoRows {
 		return err
 	}
 	return nil
 }
 
-func (c *cockroach) write(writerID string, blockNum int64, blockCount int, r *rand.Rand) error {
-	const argCount = 4
-	args := make([]interface{}, argCount*blockCount)
-	for i := 0; i < blockCount; i++ {
+func (c *cockroach) write(count int, r *rand.Rand) error {
+	const argCount = 2
+	args := make([]interface{}, argCount*count)
+	for i := 0; i < count; i++ {
 		j := i * argCount
 		args[j+0] = r.Int63()
-		args[j+1] = writerID
-		args[j+2] = blockNum + int64(i)
-		args[j+3] = randomBlock(r)
+		args[j+1] = randomBlock(r)
 	}
+	// TODO(peter): The key generation is not guaranteed unique. Consider using
+	// UPSERT, though initial tests show that is half the speed. Or perhaps
+	// ignoring duplicate key violation errors.
 	_, err := c.writeStmt.Exec(args...)
```

Comment:
I agree it needs to be addressed, but are you saying it needs to be addressed now? 64-bit random numbers makes it pretty unlikely we'll hit this for short runs. For longer runs, --tolerate-errors seems adequate for this PR. I'm working on fixing this in a follow-on PR.